tests/unit/hook_packages/terraform/hooks/prepare/prepare_base.py

```diff
@@ -467,13 +475,39 @@ def setUp(self) -> None:
             "name": "s3_lambda_code_2",
         }
 
+        self.tf_apigw_resource_properties: dict = {
+            "rest_api_id": "aws_api_gateway_rest_api.MyDemoAPI.id",
+            "parent_id": "aws_api_gateway_rest_api.MyDemoAPI.root_resource_id",
+            "path_part": "mydemoresource",
+        }
+
+        self.expected_cfn_apigw_resource_properties: dict = {
+            "RestApiId": "aws_api_gateway_rest_api.MyDemoAPI.id",
+            "ParentId": "aws_api_gateway_rest_api.MyDemoAPI.root_resource_id",
+            "PathPart": "mydemoresource",
+        }
+
+        self.tf_apigw_resource_resource: dict = {
+            **self.tf_apigw_resource_common_attributes,
+            "values": self.tf_apigw_resource_properties,
+            "address": f"aws_api_gateway_resource.{self.apigw_resource_name}",
+            "name": self.apigw_resource_name,
+        }
+
+        self.expected_cfn_apigw_resource: dict = {
+            "Type": AWS_APIGATEWAY_RESOURCE,
+            "Properties": self.expected_cfn_apigw_resource_properties,
+            "Metadata": {"SamResourceId": f"aws_api_gateway_resource.{self.apigw_resource_name}", "SkipBuild": True},
```

Comment:
`SkipBuild` property does not add any value here, can we remove it from the mapping?

Reply:
Added a condition to only set that metadata for resources that can be built.